test/react_client/js_backed_map_test.dart

```diff
@@ -23,181 +24,17 @@ main() {
       JsBackedMap jsBackedMap;
       JsBackedMap dynamicJsBackedMap;
 
-      setUp(() {
+      testTypeValue(testValue) {
```

Comment:
The type information of testValue is no longer being propagated. Are we sure that this test is equivalent to the old one? @greglittlefield-wf

Reply:
I was thinking about this and we may be able to use generics to propagate the static type information, and use JS inlining annotations from package:meta to force dart2js to use that type info (if it doesn't already):
```suggestion
      @tryInline
      void testTypeValue<T>(T testValue) {
```
I'll give that a shot locally.

Related: we should update the builder config for this repo to use the Dart 2 equivalent of the trust-type-annotations flag, since that's what we'll be using in prod most of the time.

Reply:
It looks manually inlining doesn't do much for the dart2js output, but we should add the generics so that both tests aren't essentially just `as dynamic`.

I'll make a PR to add those and also add the build.yaml and test changes